styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java

```diff
@@ -420,7 +425,7 @@ public void updateNextNaturalTrigger(WorkflowId workflowId, TriggerInstantSpec t
 
     // Strongly consistently read values for the above keys
     return Lists.partition(keys, MAX_NUMBER_OF_ENTITIES_IN_ONE_BATCH_READ).stream()
-        .map(batch -> forkJoinPool.submit(withMDC(() -> this.readRunStateBatch(batch))))
+        .map(batch -> CompletableFuture.supplyAsync(() -> this.readRunStateBatch(batch), forkJoinPool))
```

Comment:
Erm I think you meant `executor`, like in the other replaced usages of `withMDC`.

Reply:
Good catch again!